Document Segments feature: How to create and how to use in API (#586)
plausible/docs: docs/filters-segments.md

<img alt="List of subdomains" src={useBaseUrl('img/list-of-hostnames-and-subdomains.png')} />

## How to save a segment 

Click on any metric in the dashboard or use the "**Filter**" button to filter your audience to a desired segment. After completing your segment, click on the "**Save as segment**" button that is visible in the filters menu (`***`) that you see when your dashboard is filtered.

> Segments can be saved either as "**personal segments**" (these are visible only to you) or as "**site segments**" (these are visible to all the team members that have access to your dashboard).

Reviewer: One tiny note here: If the user copy pastes a dashboard link with a personal segment applied to another user (or even shares it publicly in case they are on a public dashboard), the dashboard *will load* for the others. They won't be able to see inside the segment though. So "visible" means "visible  in the dropdown". 

As an example, open this and try to edit the segment: https://pr-5129.review.plausible.io/plausible.io?f=is,segment,32&l=segment-32,Artur%27s%20segment

Choose the type of segment you want to create, give it a recognizable name and click on the "**Save**" button.

### How to open a previously saved segment

> Click on the "**Filter**" button on the top right of your dashboard to see the list of your saved segments under the "**Segments**" heading. You can also search for any of your existing saved segments. Click on a specific segment to open that segment in your dashboard. 

Reviewer 2: ```suggestion
Click the "**Filter**" button in the top-right corner of your dashboard to view your saved segments under the "**Segments**" heading. You can also search for specific segments. Click on a segment to open it in your dashboard. 
```

Note that the "**Segments**" heading will only be visible in the "**Filter**" button after you save your first segment. 

### How to edit or delete a previously saved segment

> Open a previously saved segment and then click on its name in the top line of your Plausible dashboard. Then click on the "**Edit segment**" button. Here you can add new filters or remove any existing filters from your segment. 

Reviewer 2: ```suggestion
Open a previously saved segment and click its name in the top line of your Plausible dashboard. Then, select "**Edit segment**" to add or remove filters. 
```

> You can then click on the "**Update segment**" button. Click on the arrow on the right-hand side of the "**Update segment**" button to save the existing segment as a brand new segment ("**Save as a new segment**) or to delete the existing segment ("**Delete segment**").

Reviewer 2: ```suggestion
To save your changes, click the "**Update segment**" button. Use the arrow next to this button to save the existing segment as a brand new segment ("**Save as a new segment**) or to delete the existing segment ("**Delete segment**").
```
